src/main/java/com/aws/greengrass/util/S3SdkClientFactory.java

```diff
@@ -89,33 +92,53 @@ public S3Client getS3Client() throws DeviceConfigurationException {
      * @return s3client
      */
     public S3Client getClientForRegion(Region r) {
-        setS3EndpointType(Coerce.toString(deviceConfiguration.gets3EndpointType()));
+        handleSystemPropertyUpdate();
         return clientCache.computeIfAbsent(r, (region) -> S3Client.builder()
                 .httpClientBuilder(ProxyUtils.getSdkHttpClientBuilder())
                 .serviceConfiguration(S3Configuration.builder().useArnRegionEnabled(true).build())
                 .credentialsProvider(credentialsProvider).region(r).build());
     }
 
     /**
-     * Set s3 endpoint type.
+     * Update system property and refresh client.
+     */
+    private void handleSystemPropertyUpdate() {
+        // handle s3 endpoint type
+        boolean refreshCache = handleS3EndpointType(Coerce.toString(deviceConfiguration.gets3EndpointType()));
+
+        // handle fips endpoint, check current useFips value is consistent with deviceConfig fipsMode value,
+        // refresh client if they are inconsistent
+        if (useFips.compareAndSet(Coerce.toBoolean(deviceConfiguration.getFipsMode()), useFips.get())) {
```

Comment:
Switch the args to `compareAndSet(boolean expect, boolean update)`

Reply:
```
refreshCache = useFips.compareAndSet(useFips.get(), Coerce.toBoolean(deviceConfiguration.getFipsMode()))
```

Reply:
If the fips doesn't change but s3 endpoint changed, the `refreshCache` would be false, but we do want to refresh the client in this case

Reply:
Right, you need to update that accordingly. That was a suggestion to use the compareAndSet value to refresh client. 
```
if(!isS3EndpointTypeModified(Coerce.toString(deviceConfiguration.gets3EndpointType())){
 if (!useFips.compareAndSet(useFips.get(), Coerce.toBoolean(deviceConfiguration.getFipsMode()))){
  return;
 }
}
refreshClientCache();

```

Reply:
Right, you need to update that accordingly. That was a suggestion to use the compareAndSet value to refresh client. 
```
if(isS3EndpointTypeModified(Coerce.toString(deviceConfiguration.gets3EndpointType()) ||  useFips.compareAndSet(useFips.get(), Coerce.toBoolean(deviceConfiguration.getFipsMode()))){
  refreshClientCache();
}
return;
```

Reply:
Updated